File: problem/Max_Matrix_Sum/maxMatrixSum.py
class Solution(object):
    def maxMatrixSum(self, matrix):
        totalSum = 0
        neg = 0
        minAbs = float('inf')

        for row in matrix:
            for v in row:
                if v < 0:
                    neg += 1
                av = abs(v)
                totalSum += av
                minAbs = min(minAbs, av)

        return totalSum if neg % 2 == 0 else totalSum - 2 * minAbs

File: problem/Max_Matrix_Sum/maxMatrixSum.c
long long maxMatrixSum(int **matrix, int matrixSize, int *matrixColSize)
{
    long long totalSum = 0;
    int neg = 0;
    int minAbs = INT_MAX;

    for (int i = 0; i < matrixSize; i++)
    {
        for (int j = 0; j < matrixColSize[i]; j++)
        {
            int v = matrix[i][j];
            if (v < 0)
                neg++;
            int av = abs(v);
            totalSum += av;
            if (av < minAbs)
                minAbs = av;
        }
    }

    return (neg % 2 == 0) ? totalSum : totalSum - 2LL * minAbs;
}